### PR TITLE
Add unique_id functionality. Required for further customization in HA UI

### PR DESCRIPTION
**Pull request by oskargert**
Hi, 

First of all, thank you for putting this together after they removed the funtionality from HA.
I've used your component for a couple of months without any problem. But I have been a little annoyed with not being able to customize the devices in the HA UI, such as putting them into a room, giving them a "nice" name and changing the default icon. To do this you need to add a unique id to each device. Because of this I decided to clone your repo and add the unique id functionality.

### CODE REVIEW

File: custom_components/rpi_rf/switch.py [hunk 1]
```diff
@@ -1,5 +1,7 @@
 """Support for a switch using a 433MHz module via GPIO on a Raspberry Pi."""
 from __future__ import annotations
+from email.policy import default
+from enum import unique
```
Comment (markvader) [inline custom_components/rpi_rf/switch.py:4]: not needed?
```suggestion
```

File: custom_components/rpi_rf/switch.py [hunk 2]
```diff
@@ -1,5 +1,7 @@
 """Support for a switch using a 433MHz module via GPIO on a Raspberry Pi."""
 from __future__ import annotations
+from email.policy import default
```
Comment (markvader) [inline custom_components/rpi_rf/switch.py:3]: not needed?
```suggestion
```

**Comment by markvader** [comment]
I was thinking of this as well, I have a few thoughts.
I personally want all my entities to be configurable via UI so do appreciate the pull request.

- Unique ID's such as these would not be accepted in the main project due to the ability to change the switches name in YAML after first setup.  If a switch entity's name was changed, it's configuration settings would be lost.
- https://developers.home-assistant.io/docs/entity_registry_index/ for good practice re: unique ID's
- I do like how you have combined the name together with the switches on and off codes to create a unique ID.  I was wondering what is the likelihood that a user could have two switches with the same on/off RF codes
- Also it is possible to send a string of on and off codes (i.e. a list), this would need to be handled so that only the first code was used to generate a unique ID. Again this code can be later changed in YAML so it's configuration settings would be lost in that situation.

File: custom_components/rpi_rf/switch.py (hunk 1 above)
Comment (oskargert) [reply custom_components/rpi_rf/switch.py]: Sorry, didn't realise that my linter autoimported this when I used the unique variable name

File: custom_components/rpi_rf/switch.py (hunk 2 above)
Comment (oskargert) [reply custom_components/rpi_rf/switch.py]: Sorry, didn't realise that my linter autoimported this when i used the default variable name

**Comment by oskargert** [comment]
Thank you for your feedback.

- I understand your concern with including the switch name in the unique id. I have removed this.
- In regards to the link to good practice, I would say that the on/off codes are the most unique values we have available?
- I think the likelihood for a user adding two switches with the same codes is low, but not zero. We could possibly solve this by appending a UUID identifier, but then isn't the UUID enough by itself? My concern is also that a new ID would be generated for the switch on changes to the file or restart of the server. Do you know if this would be the case?
- Would it be necessary to only use the first code? I don't see the problem with using the entire string? The changing of codes is a valid problem.

Could one solution be to document that a "manual" unique_id has to be defined in cases of using the same on/off codes for more than one switch and in the case of a switch with several codes?

In the case for a switch with several codes, the auto generation could also used simply be turned off, to force the user to define a manual unique_id?

**Comment by markvader** [comment]
Am merging into testing branch for now

**Comment by oskargert** [comment]
Great, seems like the tests worked out as expected. I agree that this is an edge case and can be solved through documentation. I'm glad I could help and contribute a little. thank you for you a good and useful component!

**Comment by markvader** [comment]
I've published the new release. Have a read of the documentation and see if you think it could be improved.

Thanks again.

**Comment by oskargert** [comment]
The documentation is good, my only comment would be that you could add the unique_id key to the configuration example and the options table, since it can be entered manually. A manually entered id would also fix the issue of keeping the same id when changing/adding the on/off codes.

**Comment by markvader** [comment]
> The documentation is good, my only comment would be that you could add the unique_id key to the configuration example and the options table, since it can be entered manually. A manually entered id would also fix the issue of keeping the same id when changing/adding the on/off codes.

Thanks. Documentation updated [commit](https://github.com/markvader/ha-rpi_rf/commit/c8dd6cccd972c3e1df94f5e4dc34a30ad22411c4)

**Comment by oskargert** [comment]
Perfect, thanks for including my feedback and letting me be a small part of your project

**Comment by markvader** [comment]
Very welcome, thanks for taking the lead on this one.
